flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -267,6 +269,22 @@ public Builder uidPrefix(String newPrefix) {
       return this;
     }
 
+    public Builder customSnapshotMetadata(Map<String, String> metadata) {
```

Comment:
How about `setSnapshotProperties` and `setSnapshotProperty`? Even those seem a bit long, but it is pretty clear.